src/game/option_passport.c

```diff
@@ -175,8 +175,12 @@ static void Option_PassportShowLevelSelect(void)
     int32_t select = DisplayRequester(&m_SelectLevelRequester);
     if (select) {
         if (select > 0) {
-            // Don't subtract 1 because level 0 is Gym
-            g_InvExtraData[IED_LEVEL_NUM] = select;
+            // If there's a gym level, first level starts at 1.
+            if (g_GameFlow.gym_level_num != -1) {
+                g_InvExtraData[IED_LEVEL_NUM] = select;
+            } else {
+                g_InvExtraData[IED_LEVEL_NUM] = select - 1;
+            }
```

Comment:
I don't quite get this, can you elaborate? @walkawayy

Reply:
Unfinished Business doesn't have a gym level, so it threw off all the calculations.

Reply:
Can it be `select - g_GameFlow.first_level_num`?

Reply:
I don't think your suggestion works.

Example with a gym (vanilla TR1):
* I load level 3 (Lost Valley) from the Select Level menu.
* `select` is 3.
* `gym_level_num` is 0.
* `first_level_num` is 1.
* `select - g_GameFlow.first_level_num` would wrongly get 2 (City of Vilcabamba).

Example without a gym (UB):
* I load level 3 (Atlantean Stronghold) from the Select Level menu.
* `select` is 3.
* `gym_level_num` is -1.
* `first_level_num` is 0.
* `select - g_GameFlow.first_level_num` would correctly get 3 (Atlantean Stronghold).

Reply:
According to what you posted above we should always assign `select` then?

Reply:
Whoops I screwed up the UB section. I edited it. Since there's no gym, level 0 is the first level. So level 3 in my example is the last (fourth) level The Hive.

Reply:
How about `select + g_GameFlow.first_level_num - 1`?

1. `select` is 3, we get 3 + 1 - 1  = 3
2. `select` is 3, we get 3 + 0 - 1 = 2

And it makes sense because we adjust for the first level number and change one-indexing to zero-indexing

Reply:
That works. Done.